src/iotHubResourceExplorer.ts

```diff
@@ -257,6 +256,26 @@ export class IoTHubResourceExplorer extends BaseExplorer {
         });
     }
 
+    private async getOrSelectSubscriptionItem(outputChannel: vscode.OutputChannel, subscriptionId: string): Promise<SubscriptionItem> {
+        if (subscriptionId) {
+            let azureResourceFilter = this.accountApi.filters.find((filter) => filter.subscription.subscriptionId === subscriptionId);

```

Comment:
is the filter for active subs or all subs?

Reply:
The filter is for active subs. Change to all subs now.